### DISCUSSION TITLE
UX/UI Suggestions

### DISCUSSION THREAD

**Discussion by newhinton** [qarmin/czkawka#1420] [Ideas]
Hi!

I am a regular(-is) user of czkawka. First, it generally does what i need, and it does it fast. That is great!

However, over the time i noticed some usability issues. I'd like to write them up here, so that others might benefit from my experience.

First my usecase:

I often find myself having two folders. A reference folder (the folder that i know contains my data) and a folder that needs sorting. (Usually a filedump, like my downloads folder). Lets call them "Reference Folder" and "Sort Folder". (There might be more than one "Sort Folder", but it generally does not matter how many of each type i have for my experiences)

Then my goal is to remove everything in the sort folder that i already have in my reference folder, since that contains my organized files. 

Since a bit of time passed since my last use of czkawka, i might have forgotten some of the details of how this app works.
This results in the following to me:

1. I add both folders via the buttons.
2. I scan for files
3. I get a list of "groups" that contain all duplicates, per duplicate.
4. There is a lot of duplicates that i might want to keep, so i have to use the "select" button below the list.

This is the first hurdle, because i often do not remember what exactly i choose last time, and only trial and error gets me what i want. (E.g. "path" does not seem to select a path-prefix, but everything, and the regex-one does what i want when i enter a path, but i'd imagine most people do not know what a regex is.)

> As a sidenote, it seems that the dividing row entries for groups in this mode have a checkbox. This checkbox seems to do nothing and is not selectable. Also clicking on images opens the sidebar with the preview, but initially this takes up 2/3 of my screen and is way to big because it hides the actual list that i need to sort

This is okay, because in the end it works as advertised and i can sort my files this way. However, i usually find myself overwhelmed by this view because (for me) it contains a lot of false positives, so i quickly try to figure out a better whay.
This is when i try to use the "reference" mode.

1. I add both folders via the buttons.
2. I select my reference folder
3. I scan for files
4. I get a list of duplicates, but no groups anymore.

This mode brings it's own challenges:
1. Since this mode does not have groups, it is hard to see how many duplicates there are of a file. 
2. It is entirely unclear what "reference" does in this mode, since the reference-files show up identically to the non-reference duplicates. The only identifying mark is that you cannot check the checkbox, and obviously their filepath.
3. "select individual" seems to do nothing.

This removes the false positves i had before, but it makes selection/identification much harder.

I'd like to make the following suggestions to improve this:

1. Create a initially fixed width for the preview-pane, independent of the actual content.
2. Remove checkboxes from rows that are not clickable
3. Maybe rework the ux of the "selection"-menu, especially the "indivdual"-menu. (Maybe it is not nessessary to have three different input fields? Wouldn't one "regex"-inputfield work aswell for all three cases?) Generally it would help if there was a "select all from this folder"-button (#1324)

Reference-Mode:
5. Introduce groups like in the other mode, with the header beeing the reference-file that is not selectable, and with bigger spacing so that the difference is also clear for "single-duplicate-entries"

There are also some more generalized things i found might help:

- Better spacing and separation of ui-elements:

The folder-selecton-pane on top is generally fine, but it feels like it clings to the window corners. Maybe it would  help to rearrange it a bit and add padding to it:

![czkawa-improvements](https://github.com/user-attachments/assets/19e3100d-d0d3-4a9e-aa51-7aacd93bccbf)
(Also moving the search button there might help with this too.)

If you select the search-constraint-tab, the "add/remove"buttons become greyed out or become hidden, but the space they take should still be occupied. I haven't added it in the edited image, but in this case the recursive-checkbox could be moved to the right-search-button-pane.

I think it might also be a good idea to move both the "check method" settings and the left pane next to the filelist into their own pane. (maybe the left pane could be a dropdown aswell?)

The last thing is the button-bar below the filelist. This buttonbar contains at least three distinct groups of actions as i see it:
list-actions, "destructive"-actions, and "app-management".
- List Actions: "Select", "Sort"
- "Destructive"-Actions: "Delete", "Move", "Symlink", "Hardlink"
- "App-Management": "Save", "Show/Hide-Bottompane",  "Show/Hide-Toppane"

Maybe the latter group could be better moved to the "gnome-style burger menu" (Sorry, i don't know how that particular ui interface is called. But it is usually a drop-down menu which is located where currently the app-settings and the app-info buttons are in the navbar)

The other two sets of buttons are very important, and i think they deserve to be a bit more prominent, though i'd suggest to seperate them a bit more to create a distinct differentiation between them. 

Let me know what you think! I might give some things a shot myself if you want that!

**Comment by newhinton** [comment]
![image](https://github.com/user-attachments/assets/c55d605c-9182-4420-870e-9dc8de00c80f)

This is roughly what i mean. There is obviously a lot wrong, and the vertical-tab-bar is really a step-down ux-wise from the current state, but i imagine that using a dropdown menu that then shows the respective settings, in combination with a dedicated file-list would really clean that up too. However, that is outside of my current skill level, so i'd first like to hear some suggestions before learning how to do this.

**Comment by bandaid** [comment]
Yes, missing SORTING ability on Windows version too. Sometimes I just want to bulk select duplicates and delete, instead I must tediously select every other item and delete...
